src/core/mempool.rs

```diff
@@ -18,18 +63,411 @@ pub struct MempoolAdmitter {
     cur_burn_block: BurnchainHeaderHash,
 }
 
-impl MempoolAdmitter {
-    pub fn new(chainstate: StacksChainState, cur_block: BlockHeaderHash, cur_burn_block: BurnchainHeaderHash) -> MempoolAdmitter {
-        MempoolAdmitter { chainstate, cur_block, cur_burn_block }
+impl MemPoolAdmitter {
+    pub fn new(chainstate: StacksChainState, cur_block: BlockHeaderHash, cur_burn_block: BurnchainHeaderHash) -> MemPoolAdmitter {
+        MemPoolAdmitter { chainstate, cur_block, cur_burn_block }
     }
 
     pub fn set_block(&mut self, cur_block: &BlockHeaderHash, cur_burn_block: &BurnchainHeaderHash) {
         self.cur_burn_block = cur_burn_block.clone();
         self.cur_block = cur_block.clone();
     }
 
-    pub fn will_admit_tx<R: Read>(&mut self, tx: &mut R) -> Result<StacksTransaction, MemPoolRejection> {
-        self.chainstate.will_admit_mempool_tx(&self.cur_burn_block, &self.cur_block, tx)
+    pub fn will_admit_tx(&mut self, tx: &StacksTransaction, tx_size: u64) -> Result<(), MemPoolRejection> {
+        self.chainstate.will_admit_mempool_tx(&self.cur_burn_block, &self.cur_block, tx, tx_size)
+    }
+}
+
+pub struct MemPoolTxInfo {
+    pub tx: StacksTransaction,
+    pub len: u64,
+    pub fee: u64,
+    pub burn_header_hash: BurnchainHeaderHash,
+    pub block_header_hash: BlockHeaderHash,
+    pub block_height: u64,
+    pub accept_time: u64,
+}
+
+impl FromRow<MemPoolTxInfo> for MemPoolTxInfo {
+    fn from_row<'a>(row: &'a Row) -> Result<MemPoolTxInfo, db_error> {
+        let txid = Txid::from_column(row, "txid")?;
+        let tx_bytes : Vec<u8> = row.get("tx");
+        let burn_header_hash = BurnchainHeaderHash::from_column(row, "burn_header_hash")?;
+        let block_header_hash = BlockHeaderHash::from_column(row, "block_header_hash")?;
+        let fee_i64 : i64 = row.get("fee");
```

Comment:
I think implementing FromSql and ToSql for u64 is preferable to this approach.

Reply:
Yes; there's already an open issue for this.

Reply:
Yep -- but I think that because this PR adds several new instances of u64s as i64s, that it should probably be done in this PR.

Reply:
So, I don't think we can do this directly.  `ToSql` and `FromSql` are traits from a different crate, which precludes implementing them for arbitrary types.  We'd have to wrap `u64` into a Stacks-specific struct, implement `ToSql` and `FromSql` for that struct, and then implement a destruct method for that struct to get the inner `u64`.  I'm not sure this would be an improvement.

What if instead, I implemented `FromRow<T>` and `FromColumn<T>` for `u64`, so then we can do `let fee = u64::from_column("fee")?;`

Reply:
Fixed in 44fa503 and 0cbbfb1

Reply:
I see -- another option would be to define a newtype like `struct u64_sql(i64)`, and define the ToSql and FromSql traits for that type. The benefit of a ToSql impl is that it can check and error when supplied a u64 > 2^63. Currently, this will lead to a database corruption, but the error is not surfaced until the value is read, which makes debugging difficult, but also at worst, may mean that the error goes undetected for some stretch of time. Alternatively, we could just add a function u64_to_sql to handle the conversion (and yield an error).

Reply:
> I see -- another option would be to define a newtype like struct u64_sql(i64), and define the ToSql and FromSql traits for that type. 

Yup; that's what I described above.  I see your point about the need to guard writes; I'll either take this approach or add a `u64_to_sql` method.

Reply:
Fixed in 51035e9